Add memberOf flag output to classes and interfaces. Enabled jump to description in inserted comment.

mredbishop opened the pull request:
Added support for memberOf flag to allow better documentation of class properties and methods.

mredbishop commented:
Added a tweak to leave the cursor at the description location of the inserted comment for single inserts.

joelday commented on src/documenter.ts:
```diff
@@ -202,6 +213,10 @@ export class Documenter implements vs.Disposable {
             }
         }
+
+        if (goToDescription) setTimeout(() => {
```
Subjective nitpick: use braces here

joelday commented on src/documenter.ts:
```diff
@@ -202,6 +213,10 @@ export class Documenter implements vs.Disposable {
+
+        if (goToDescription) setTimeout(() => {
+            this._goToDescriptionLocation(commentText)
+        }, 100);
```
I wish there was a different way to do this, but if it works, it works. Can you do a quick test of this with a huge ass file?

joelday commented:
Can you add a setting to disable memberOf output? Would probably be ideal to set it on or off for everything/interfaces/enums. Only interfaces and enums by default?

Thanks a ton for doing all of this!

joelday commented:
Oh, also, feel free to rev the version, update the readme and credit yourself for contributing.

mredbishop commented:
No worries, thanks for all the rest of the extension :+1:  

What would you like the options to be exactly for memberOf? We use it most for classes as the JSDoc generator doesn't show the documentation for the class members without it. 

Is there a **good** tool for generating the docs directly from the typescript?

mredbishop commented:
**As per pull request comments:**

- Added in curly braces as noted in comment. 
- Tested description cursor move on a massive file (4000+ lines, 1000 functions) and it works great. 
- Did a load of other testing using largeFile.ts. 
- Added @memberOf note to the readme.md.
- Added configuration options to turn all this stuff off.
- Updated the version number.
- Added my info to the end of the contributors list.

Ed

joelday commented:
In my experience, the best bet is to compile to ES6 when documenting, so JSDoc is able to infer from the class declarations itself. It won't pick up enum members though, so those need it either way. I don't think the TypeScript output puts the comments in the right place, unfortunately.

Maybe you can try transforming TypeDoc's raw data to JSDoc's raw data? Then, you wouldn't need any type annotation either.

joelday commented:
Published!

mredbishop commented:
Great, thanks Joel. I've installed that from the marketplace now and it's
all working as expected.

Let me know if there are any features bugs that you want looking at. We
have a bit of spare time here and there to work on open source stuff so
might be able to help out.

Cheers,

Ed

On Wed, 7 Sep 2016 at 21:59 Joel Day <notifications@github.com> wrote:

> Published!
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/joelday/vscode-docthis/pull/45#issuecomment-245417167>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABC6ZTOLW9QD_RdM1rx1eElEaqoiJ8xVks5qnyWegaJpZM4J2xoI>
> .
>
